plans/gmail-oauth-authentication.md

```diff
@@ -0,0 +1,1956 @@
+- `GET /v1/wallets/{walletId}` - Get wallet details (owner only)
+- `GET /v1/orders/{orderId}` - Get order details (owner only)
+- `GET /v1/orders` - List user's orders (authenticated users)
+
+**Service Deployment Architecture:**
+```
+┌─────────────────┐
+│  Auth Service   │ - Handles OAuth + JWT generation
+│  /v1/auth/*     │
+└─────────────────┘
+         │
+         │ Issues JWT
+         ▼
+┌─────────────────┐
+│ Commands Service│ - Validates JWT, processes commands
+│  /v1/accounts   │
+│  /v1/wallets    │
+│  /v1/orders     │
+└─────────────────┘
+         │
+         │ Emits events
+         ▼
+┌─────────────────┐
+│ Queries Service │ - Validates JWT, serves queries
+│  /v1/accounts   │
+│  /v1/wallets    │
+│  /v1/orders     │
+└─────────────────┘
+```
+
+#### 8. Frontend Integration
+
+The frontend will:
+1. Redirect users to `AUTH_SERVICE_URL/v1/auth/login` to start OAuth flow
+2. Receive JWT tokens from `/v1/auth/callback` after successful authentication
+3. Store access token and refresh token securely (memory or secure storage)
+4. Include access token in `Authorization: Bearer <token>` header for all API requests to Commands/Queries services
+5. Implement token refresh logic when access token expires (401 response) by calling `AUTH_SERVICE_URL/v1/auth/refresh`
+6. Call `/v1/auth/user` with JWT to get user profile
+7. Handle authentication errors and redirect to login when refresh token expires
+
+**Environment Configuration:**
+```properties
+# Frontend Configuration
+AUTH_SERVICE_URL=http://localhost:8080
+COMMANDS_SERVICE_URL=http://localhost:8081
+QUERIES_SERVICE_URL=http://localhost:8082
+```
+
+### Extensibility for Other Providers
+
+The architecture is designed for easy extension:
+
+1. **Provider-Specific User Info Classes**: Implement `OAuth2UserInfo` interface
+2. **Provider Configuration**: Add new registration in application.yml
+3. **Provider Detection**: Use registration ID to select appropriate user info extractor
+4. **Common Service Layer**: `OAuth2UserService` handles all providers uniformly
+
+**Example for GitHub:**
+```yaml
+spring:
+  security:
+    oauth2:
+      client:
+        registration:
+          github:
+            client-id: ${GITHUB_CLIENT_ID}
+            client-secret: ${GITHUB_CLIENT_SECRET}
+            scope:
+              - user:email
+              - read:user
+```
+
+```java
+public class GitHubOAuth2UserInfo implements OAuth2UserInfo {
+    // GitHub-specific attribute mapping
+}
+```
+
+## Implementation Steps
+
+### Phase 1: Foundation Setup
+1. **Add Maven Dependencies** to `commands/pom.xml` and `queries/pom.xml`:
+   - `spring-boot-starter-security`
+   - `spring-boot-starter-oauth2-client`
+   - `spring-security-oauth2-resource-server` (for JWT validation via Nimbus)
+   - `spring-security-oauth2-jose` (includes Nimbus JOSE+JWT library)
+   - `google-auth-library-oauth2-http` (for GCP Service Account JWT signing)
+
+2. **Create GCP Service Account**:
```

Comment:
The service account is created in another (terraform) repo and should be a given here. Please change to "ensure GCP serivce acccount is created"